tests/typechecker/return-value-type-mismatch.error

```diff
@@ -0,0 +1 @@
+Parameter type mismatch: String vs u64
```

Comment:
I don't like this error message, which type is it expecting? How about something along the lines of:
`Expected "Foo", got "Bar"`

Reply:
Yeah, I agree. I had just propagated the error message up from `check_types_for_compat`, but since that function is used in contexts other than function calls anyway, I changed the error message there to a clearer and more general type mismatch error message, as suggested. Thanks for the review!